packages/agent-builder/src/workflows/template-builder.ts

```diff
@@ -0,0 +1,1682 @@
+          const pascalCaseCount = tsFiles.filter(f => /^[A-Z][a-zA-Z0-9]*\.ts$/.test(f)).length;
+
+          const max = Math.max(camelCaseCount, snakeCaseCount, kebabCaseCount, pascalCaseCount);
+          if (max === 0) return 'unknown';
+
+          if (camelCaseCount === max) return 'camelCase';
+          if (snakeCaseCount === max) return 'snake_case';
+          if (kebabCaseCount === max) return 'kebab-case';
+          if (pascalCaseCount === max) return 'PascalCase';
+
+          return 'unknown';
+        } catch {
+          return 'unknown';
+        }
+      };
+
+      // Convert naming based on convention
+      const convertNaming = (name: string, convention: string): string => {
+        const baseName = basename(name, extname(name));
+        const ext = extname(name);
+
+        switch (convention) {
+          case 'camelCase':
+            return (
+              baseName
+                .replace(/[-_]/g, '')
+                .replace(/([A-Z])/g, (match, p1, offset) => (offset === 0 ? p1.toLowerCase() : p1)) + ext
+            );
+          case 'snake_case':
+            return (
+              baseName
+                .replace(/[-]/g, '_')
+                .replace(/([A-Z])/g, (match, p1, offset) => (offset === 0 ? '' : '_') + p1.toLowerCase()) + ext
+            );
+          case 'kebab-case':
+            return (
+              baseName
+                .replace(/[_]/g, '-')
+                .replace(/([A-Z])/g, (match, p1, offset) => (offset === 0 ? '' : '-') + p1.toLowerCase()) + ext
+            );
+          case 'PascalCase':
+            return baseName.replace(/[-_]/g, '').replace(/^[a-z]/, match => match.toUpperCase()) + ext;
+          default:
+            return name;
+        }
+      };
+
+      // Process each unit
+      for (const unit of orderedUnits) {
+        console.log(`Processing ${unit.kind} unit "${unit.id}" from file "${unit.file}"`);
+
+        // Resolve source file path with fallback logic
+        let sourceFile: string;
+        let resolvedUnitFile: string;
+
+        // Check if unit.file already contains directory structure
+        if (unit.file.includes('/')) {
+          // unit.file has path structure (e.g., "src/mastra/agents/weatherAgent.ts")
+          sourceFile = resolve(templateDir, unit.file);
+          resolvedUnitFile = unit.file;
+        } else {
+          // unit.file is just filename (e.g., "weatherAgent.ts") - use fallback
+          const folderPath =
+            AgentBuilderDefaults.DEFAULT_FOLDER_STRUCTURE[
+              unit.kind as keyof typeof AgentBuilderDefaults.DEFAULT_FOLDER_STRUCTURE
+            ];
+          if (!folderPath) {
+            conflicts.push({
+              unit: { kind: unit.kind, id: unit.id },
+              issue: `Unknown unit kind: ${unit.kind}`,
+              sourceFile: unit.file,
+              targetFile: 'N/A',
+            });
+            continue;
+          }
+          resolvedUnitFile = `${folderPath}/${unit.file}`;
```

Comment:
I also think all this logic will fail if their repo structure is slightly different than what we expect